plugin/trino-snowflake/src/test/java/io/trino/plugin/snowflake/TestSnowflakeTypeMapping.java

```diff
@@ -348,6 +350,34 @@ private void testTimestamp(ZoneId sessionZone)
                 .execute(getQueryRunner(), session, trinoCreateAndInsert("test_timestamp"));
     }
 
+    @Test
+    public void testTimestampTZ()
+    {
+        testTimestampTZ(UTC);
+        testTimestampTZ(jvmZone);
+        testTimestampTZ(vilnius);
+        testTimestampTZ(kathmandu);
+        testTimestampTZ(TestingSession.DEFAULT_TIME_ZONE_KEY.getZoneId());
+    }
+
+    private void testTimestampTZ(ZoneId sessionZone)
+    {
+        Session session = Session.builder(getSession())
+                .setTimeZoneKey(TimeZoneKey.getTimeZoneKey(sessionZone.getId()))
+                .build();
+        SqlDataTypeTest.create()
+                .addRoundTrip("TIMESTAMP_TZ(0)", "'2020-09-27 12:34:56 -08:00'::TIMESTAMP_TZ", createTimestampWithTimeZoneType(0), "TIMESTAMP '2020-09-27 12:34:56 -08:00'")
+                .addRoundTrip("TIMESTAMP_TZ(3)", "'2020-09-27 12:34:56.333 -08:00'::TIMESTAMP_TZ", createTimestampWithTimeZoneType(3), "TIMESTAMP '2020-09-27 12:34:56.333 -08:00'")
+                .addRoundTrip("TIMESTAMP_TZ(9)", "'2020-09-27 12:34:56.999999999 -08:00'::TIMESTAMP_TZ", createTimestampWithTimeZoneType(9), "TIMESTAMP '2020-09-27 12:34:56.999999999 -08:00'")
+                .execute(getQueryRunner(), session, snowflakeCreateAndInsert("tpch.test_timestamptz_sf_insert"));
+        SqlDataTypeTest.create()
+                .addRoundTrip("TIMESTAMP(0) WITH TIME ZONE", "TIMESTAMP '2020-09-27 12:34:56 -08:00'", createTimestampWithTimeZoneType(0), "TIMESTAMP '2020-09-27 12:34:56 -08:00'")
+                .addRoundTrip("TIMESTAMP(3) WITH TIME ZONE", "TIMESTAMP '2020-09-27 12:34:56.333 -08:00'", createTimestampWithTimeZoneType(3), "TIMESTAMP '2020-09-27 12:34:56.333 -08:00'")
+                .addRoundTrip("TIMESTAMP(9) WITH TIME ZONE", "TIMESTAMP '2020-09-27 12:34:56.999999999 -08:00'", createTimestampWithTimeZoneType(9), "TIMESTAMP '2020-09-27 12:34:56.999999999 -08:00'")
```

Comment:
Please add more test cases, e.g. null, min, max, julian->gregorian calendar switch and etc. 
You can refer to TestPostgreSqlTypeMapping class.

Reply:
Ack. I just read the existing test cases in `TestSnowflakeTypeMapping::testDate`. It's testing min/max values like `0001-01-01` and `99999-12-31`. However, years beyond the range `1582~9999` are not recommended by Snowflake: https://docs.snowflake.com/en/sql-reference/data-types-datetime#date. Do we actually want to remove them? Also, I feel it's best to test Gregorian Calendar with leap days like `2024-02-29`. I can actually create a separate PR to clean this up, altogether with `testDate`, `testTimestamp`, and `testTimestampWithTimeZone`. What do you think?